docs/source/deployment/databricks/databricks_deployment_workflow.md

```diff
@@ -0,0 +1,311 @@
+
+```bash
+kedro package
+```
+
+This command generates a `.whl` file in the `dist` directory within your project's root directory.
+
+### Upload project data and configuration to DBFS
+
+```{note}
+A Kedro project's configuration and data do not get included when it is packaged. They must be stored somewhere accessible to allow your packaged project to run.
+```
+
+Your packaged Kedro project needs access to data and configuration in order to run. Therefore, you will need to upload your project's data and configuration to a location accessible to Databricks. In this guide, we will store the data on the Databricks File System (DBFS).
+
+The Databricks Iris starter contains a [catalog](../../data/data_catalog.md#the-data-catalog) that is set up to access data stored in DBFS (`<project_root>/conf/`). You will point your project to use configuration stored on DBFS using the `--conf-source` option when you create your job on Databricks.
+
+There are several ways to upload data to DBFS: you can use the [DBFS API](https://learn.microsoft.com/en-us/azure/databricks/dev-tools/api/latest/dbfs), the [`dbutils` module](https://docs.databricks.com/dev-tools/databricks-utils.html) in a Databricks notebook or the [Databricks CLI](https://docs.databricks.com/dev-tools/cli/dbfs-cli.html). In this guide, it is recommended to use the Databricks CLI because of the convenience it offers.
+
+- **Upload your project's data and config**: at the command line in your local environment, use the following Databricks CLI commands to upload your project's locally stored data and configuration to DBFS:
+
+```bash
+databricks fs cp --recursive <project_root>/data/ dbfs:/FileStore/iris-databricks/data
+databricks fs cp --recursive <project_root>/conf/ dbfs:/FileStore/iris-databricks/conf
+```
+
+The `--recursive` flag ensures that the entire folder and its contents are uploaded. You can list the contents of the destination folder in DBFS using the following command:
+
+```bash
+databricks fs ls dbfs:/FileStore/iris-databricks/data
+```
+
+You should see the contents of the project's `data/` directory printed to your terminal:
+
+```bash
+01_raw
+02_intermediate
+03_primary
+04_feature
+05_model_input
+06_models
+07_model_output
+08_reporting
+```
+
+## Deploy and run your Kedro project using the workspace UI
+
+To run your packaged project on Databricks, login to your Databricks account and perform the following steps in the workspace:
+
+1. **Create a new job**: In the Databricks workspace, navigate to the `Worfklows` tab and click `Create Job`:
+
+![Create Databricks job](../../meta/images/databricks_create_new_job.png)
+
+2. **Create a new job cluster**: Create a dedicated [job cluster](https://docs.databricks.com/clusters/index.html) to run your job by clicking on the drop-down menu in the `Cluster` field:
```

Comment:
"If you don't have a cluster yet" or something like that?

Reply:
Databricks will provide you with a 'default' job cluster, though you shouldn't use this to run your job and you should properly configure one for this job. I have added a sentence to make sure the user understands this.

In addition, this job cluster is far overpowered for the demands of `databricks-iris` and running it will incur a lot of unnecessary cost.

This is why your job run failed, the default job cluster tries to load some secrets that we haven't set, giving you the error you saw:

<img width="626" alt="Screenshot 2023-05-30 at 12 25 09" src="https://github.com/kedro-org/kedro/assets/37243923/5a5eb173-8542-4a63-913b-36ac6f7d4de5">

Reply:
Weird - even creating a new cluster makes this fail. Right after "add new cluster", if I navigate to Advanced I'll see these environment variables.

```
PYSPARK_PYTHON=/databricks/python3/bin/python3
PIP_EXTRA_INDEX_URL_USER={{secrets/private/PIP_EXTRA_INDEX_URL_USER}}
PIP_EXTRA_INDEX_URL_SECRET={{secrets/private/PIP_EXTRA_INDEX_URL_SECRET}}
```

If I try to remove them, the following validation error appears:

![image](https://github.com/kedro-org/kedro/assets/316517/3bfd11b2-edcf-4535-ad9e-c80d5cb19580)

Maybe I'm missing something?